packages/react-interactions/events/src/dom/Focus.js

```diff
@@ -45,28 +45,25 @@ type FocusProps = {
   onFocusVisibleChange: boolean => void,
 };
 
-type FocusEventType =
-  | 'focus'
-  | 'blur'
-  | 'focuschange'
-  | 'focusvisiblechange'
-  | 'detachedvisiblenode';
+type FocusEventType = 'focus' | 'blur' | 'focuschange' | 'focusvisiblechange';
 
 type FocusWithinProps = {
   disabled?: boolean,
   onFocusWithin?: (e: FocusEvent) => void,
   onBlurWithin?: (e: FocusEvent) => void,
   onFocusWithinChange?: boolean => void,
   onFocusWithinVisibleChange?: boolean => void,
-  onDetachedVisibleNode?: (e: FocusEvent) => void,
+  onBeforeFocusedElementDetached?: (e: FocusEvent) => void,
+  onFocusedElementDetached?: (e: FocusEvent) => void,
```

Comment:
This wouldn't be needed, `onBlurWithin` payload can include the `elementAttached` data